lib/ravelin/event.rb

```diff
@@ -60,14 +61,14 @@ def format_timestamp(timestamp)
     def validate_top_level_payload_params
       validate_customer_id_presence_on :order, :paymentmethod,
         :pretransaction, :transaction, :label
-      case name
+      case @internal_name
       when :customer
         validate_payload_inclusion_of :customer
       when :voucher
         validate_payload_inclusion_of :voucher
       when :'paymentmethod/voucher'
         validate_payload_inclusion_of :'voucher_redemption'
-      when :pretransaction, :transaction
+      when :pre_transaction, :transaction
```

Comment:
What caused this change? Was it wrong in the first place, or is it an API version thing?

Reply:
`@internal_name` contains the unconverted name provided to the `initialize` method.

To disambiguate between the v2 login event (`login`) and v3 login event (`ato_login`), I had to switch to using the `@internal_name` variable instead of `name` in this case statement to apply the correct validation to each event. 

The change here is to reference the pre-converted value.

```ruby
    def convert_event_name(str)
      underscore_mapping = {
        payment_method: :paymentmethod,
        pre_transaction: :pretransaction,
        ato_login: :login,
        ato_reclaim: :reclaim,
        label: 'label/customer'.to_sym
      }
```